drivers/gpio/gpio_dw.c

```diff
@@ -205,28 +302,81 @@ static inline void dw_port_config(struct device *port, int flags)
 static inline int gpio_dw_config(struct device *port, int access_op,
 				 u32_t pin, int flags)
 {
+	const struct gpio_dw_config *config = port->config->config_info;
+	struct gpio_dw_runtime *context = port->driver_data;
+	u32_t port_base_addr = context->base_addr;
+	u32_t data_port = dw_get_data_port(port_base_addr);
+	u32_t io_flags;
+
+	/* Check for invalid pin number */
+	if (pin >= config->bits) {
+		return -EINVAL;
+	}
+
+	/* Does not support disconnected pin, and
+	 * not supporting both input/output at same time.
+	 */
+	io_flags = flags & (GPIO_INPUT | GPIO_OUTPUT);
+	if ((io_flags == GPIO_DISCONNECTED)
+	    || (io_flags == (GPIO_INPUT | GPIO_OUTPUT))) {
+		return -ENOTSUP;
+	}
+
+	/* Check for an invalid pin configuration */
+	if (((flags & GPIO_INT_ENABLE) != 0U) &&
+	    ((flags & GPIO_INPUT) == 0U)) {
+		/* Interrupt enabled but pin is not set to input */
+		return -EINVAL;
+	}
+
+	/* Only PORT-A support interrupts and de-bounce */
+	if (data_port != SWPORTA_DR) {
+		if ((flags & GPIO_INT_ENABLE) != 0U) {
+			return -ENOTSUP;
+		}
+
+		if ((flags & GPIO_INT_DEBOUNCE) != 0U) {
+			return -ENOTSUP;
+		}
+	}
+
+	/* Cannot be output and interrupt enabled */
 	if ((flags & GPIO_INT) && (flags & GPIO_DIR_OUT)) {
```

Comment:
There is already check on line 326 (and we know that simultaneous input/output is not supported).